sentry-samples/sentry-samples-console/src/main/java/io/sentry/samples/console/Main.java

```diff
@@ -0,0 +1,132 @@
+                    && breadcrumb.getMessage().contains("bad breadcrumb")) {
+                  return null;
+                }
+                return breadcrumb;
+              });
+
+          // Configure the background worker which sends events to sentry:
+          // Wait up to 5 seconds before shutdown while there are events to send.
+          options.setShutdownTimeout(5000);
+
+          // Enable SDK logging with Debug level
+          options.setDebug(true);
+          // To change the verbosity, use:
+          options.setDiagnosticLevel(SentryLevel.LOG);
+
+          // Using a proxy:
+          options.setProxy(null); // new Proxy(Type.HTTP, new InetSocketAddress(8090)));
+
+          // Exclude classes from certain packages from stack traces sent to Sentry:
+          options.addInAppExclude("org.jboss");
+        });
+
+    Sentry.addBreadcrumb(
+        "A 'bad breadcrumb' that will be rejected because of 'BeforeBreadcrumb callback above.'");
+
+    // Data added to the root scope (no PushScope called up to this point)
+    // The modifications done here will affect all events sent and will propagate to child scopes.
+    Sentry.configureScope(
+        scope -> {
+          scope.addEventProcessor(new SomeEventProcessor());
+
+          scope.setExtra("SomeExtraInfo", "Some value for extra info");
+        });
+
+    // Configures a scope which is only valid within the callback
+    Sentry.withScope(
+        scope -> {
+          scope.setLevel(SentryLevel.FATAL);
+          scope.setTransaction("main");
+
+          Sentry.captureMessage("Fatal message!");
+        });
+
+    Sentry.captureMessage("Some warning!", SentryLevel.WARNING);
+
+    Exception exception = new RuntimeException("Attempting to send this multiple times");
+
+    // Only the first capture will be sent to Sentry
+    for (int i = 0; i < 3; i++) {
+      // The SDK is able to detect duplicate events:
+      // This is useful, for example, when multiple loggers log the same exception. Or exception is
+      // re-thrown and recaptured.
+      Sentry.captureException(exception);
+    }
+
+    int count = 10;
+    for (int i = 0; i < count; i++) {
+      String msg = "%d of %d items we'll wait to flush to Sentry!";
+      Message message = new Message();
+      message.setMessage(msg);
+      message.setFormatted(String.format(msg, i, count));
+      SentryEvent event = new SentryEvent();
+      event.setMessage(message);
+      Sentry.captureEvent(event, SentryLevel.DEBUG);
+    }
+
+    // Console output will show queue being flushed. Task completes then and timeout is never
+    // reached (you don't need to wait a day :)
+    Sentry.flush(10000);
+
+    // An event with breadcrumb and user data
+    SentryEvent evt = new SentryEvent();
+    Message msg = new Message();
+    msg.setMessage("Detailed event");
+    evt.setMessage(msg);
+    evt.addBreadcrumb(new Breadcrumb("Breadcrumb directly to the event"));
```

Comment:
I'd love for us to have an overload like this:

```suggestion
    evt.addBreadcrumb("Breadcrumb directly to the event");
```

Reply:
I'll add it in separate PR.

Reply:
https://github.com/getsentry/sentry-android/pull/500

Reply:
PR is approved.
although the Sentry static class has overloads for adding breadcrumbs like this, also a few others overload.
new SentryEvent() and calling adders and setters were never intended to be a public API iirc, so not sure if it's worth the effort of adding methods to this class, most probably users will be doing Sentry.addBreadcrumb(...)